libraries/core/src/main/java/com/paritytrading/philadelphia/FIXMessage.java

```diff
@@ -130,7 +130,20 @@ public void setFieldCount(int count) {
      *   of a field with the specified tag
      */
     public FIXValue valueOf(int tag) {
-        for (int i = 0; i < count; i++) {
+        return valueOf(tag, 0);
+    }
+
+    /**
+     * Get the value container of the first instance of a field with the
+     * specified tag, starting at the specified index.
+     *
+     * @param tag the tag
+     * @param fromIndex the index to start the search from.
+     * @return the value container or {@code null} if there are no instances
+     *   of a field with the specified tag
+     */
+    public FIXValue valueOf(int tag, int fromIndex) {
+        for (int i = fromIndex; i < count; i++) {
```

Comment:
In the `#valueOf(int,int)` method, we could simplify the resulting bytecode slightly by eliminating `i` and using `fromIndex` as the index variable within the loop. As in:
```java
for (; fromIndex < count; fromIndex++) {
    ...
```